_episodes/21-section3-intro.md

```diff
@@ -0,0 +1,26 @@
+---
+title: "Section 3: Software Architecture and Design"
+colour: "#f7d77e"
+start: true
+teaching: 5
+exercises: 0
+questions:
+- "What architectural and software design paradigms exist to help us design our software?"
+objectives:
+- "TODO"
+keypoints:
+- "TODO"
+---
+In this section, we step away from writing code for a bit to look into different software design paradigms to understand the wider landscape in which you can design software and to help you with the design decisions you have to make. Each paradigm represents a slightly different way of thinking about and structuring the code and each has certain strengths and weaknesses when used to solve particular types of problems - useful to know so you can recognise where it might be useful to switch in your own code.
```

Comment:
"understand the wider landscape" - seems a bit too broad. Perhaps "understand a broader set of approaches you can take to design software..."

Reply:
Fixed via 7680eeb.